src/platform/site-wide/side-nav/components/NavItem.js

```diff
@@ -2,53 +2,42 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { get } from 'lodash';
+import classNames from 'classnames';
 // Relative
 import NavItemRow from './NavItemRow';
 import { NavItemPropTypes } from '../prop-types';
 
-/*
- Detects expanded children to modify the ending line style
- with a short delay.
- This is looks a bit tricky but ending line will have to render always first
- for level 1 items and then because of the recursive way of rendering children
- after this we can detect that children has been expanded and apply
- the style by finding it by its parentID.
-*/
-const shouldModifyEndingLine = (item, depth) => {
-  const expanded = get(item, 'expanded');
-  const hasChildren = get(item, 'hasChildren');
-  const shouldShowLineOpen = !!(depth === 2 && expanded && hasChildren);
-  if (shouldShowLineOpen) {
-    setTimeout(() => {
-      const element = document.getElementById(`${item.parentID}-line`);
-      if (element) {
-        element.className = 'line-open';
-      }
-    }, 1);
-  }
-  return null;
-};
-
-const NavItem = ({
-  depth,
-  item,
-  index,
-  renderChildItems,
-  sortedNavItems,
-  toggleItemExpanded,
-}) => {
+const NavItem = ({ depth, item, renderChildItems, toggleItemExpanded }) => {
   // Derive the item properties.
   const expanded = get(item, 'expanded');
   const hasChildren = get(item, 'hasChildren');
   const id = get(item, 'id');
+  const isSelected = get(item, 'isSelected');
+  // Expanded not selected
+  const isExpanded = get(item, 'expanded') && depth === 2 && !isSelected;
+  // Expanded beyond level 2 expanded and selected
+  const moreThanLevel2SelectedExpanded =
+    get(item, 'expanded') && depth > 2 && isSelected;
 
-  // Derive the depth booleans.
-  const isFirstLevel = depth === 1;
+  const isExpandedNotSelected = !!(
+    isExpanded &&
+    !moreThanLevel2SelectedExpanded &&
+    hasChildren
+  );
 
-  // Determine if we are the last nav item.
-  const isLastNavItem = index === sortedNavItems.length - 1;
   return (
-    <li className={`va-sidenav-level-${depth}`} key={id}>
+    <li
+      className={classNames(`va-sidenav-level-${depth}`, {
+        active: isSelected,
+        selected:
+          (!isExpanded &&
```

Comment:
NIT: I think it would be more readable if the `selected` condition was extracted into a variable. 

```
const shouldHaveSelectedClassName = !isExpanded && !moreThanLevel2SelectedExpanded && hasChildren && isSelected) || isExpandedNotSelected
return (
  <li 
      key={id}
      className={classNames(`va-sidenav-level-${depth}`, {
        active: isSelected,
        selected: shouldHaveSelectedClassName
      })
  >
    ...
```

Reply:
Agreed, @billfienberg  - update